cl/persistence/format/snapshot_format/getters/execution_snapshot.go

```diff
@@ -107,60 +80,31 @@ func convertWithdrawalsToBytesSSZ(ws []*types.Withdrawal) []byte {
 	return ret
 }
 
-func (r *ExecutionSnapshotReader) WithdrawalsSZZ(w io.Writer, number uint64, hash libcommon.Hash) error {
-	ok, err := r.lookupWithdrawalsInCache(w, number, hash)
-	if err != nil {
-		return err
-	}
-	if ok {
-		return nil
-	}
+func (r *ExecutionSnapshotReader) Withdrawals(number uint64, hash libcommon.Hash) (*solid.ListSSZ[*cltypes.Withdrawal], error) {
 	tx, err := r.db.BeginRo(r.ctx)
 	if err != nil {
-		return err
+		return nil, err
 	}
 	defer tx.Rollback()
 	// Get the body and fill both caches
-	body, err := r.blockReader.BodyWithTransactions(r.ctx, tx, hash, number)
+	body, _, err := r.blockReader.Body(r.ctx, tx, hash, number)
 	if err != nil {
-		return err
+		return nil, err
 	}
 	if body == nil {
-		return fmt.Errorf("transactions not found for block %d", number)
-	}
-	// compute txs flats
-	txs, err := types.MarshalTransactionsBinary(body.Transactions)
-	if err != nil {
-		return err
-	}
-	flattenedTxs := convertTxsToBytesSSZ(txs)
-	r.txsCache.Add(cacheEntry{number: number, hash: hash}, flattenedTxs)
-	// compute withdrawals flat
-	ws := body.Withdrawals
-	flattenedWs := convertWithdrawalsToBytesSSZ(ws)
-
-	r.withdrawalsCache.Add(cacheEntry{number: number, hash: hash}, flattenedWs)
-	_, err = w.Write(flattenedWs)
-
-	return err
-}
-
-func (r *ExecutionSnapshotReader) lookupWithdrawalsInCache(w io.Writer, number uint64, hash libcommon.Hash) (bool, error) {
-	var wsBytes []byte
-	var ok bool
-	if wsBytes, ok = r.withdrawalsCache.Get(cacheEntry{number: number, hash: hash}); !ok {
-		return false, nil
-	}
-	_, err := w.Write(wsBytes)
-	return true, err
-}
-
-func (r *ExecutionSnapshotReader) lookupTransactionsInCache(w io.Writer, number uint64, hash libcommon.Hash) (bool, error) {
-	var wsBytes []byte
-	var ok bool
-	if wsBytes, ok = r.txsCache.Get(cacheEntry{number: number, hash: hash}); !ok {
-		return false, nil
-	}
-	_, err := w.Write(wsBytes)
-	return true, err
+		return nil, fmt.Errorf("transactions not found for block %d", number)
+	}
+	ret := solid.NewStaticListSSZ[*cltypes.Withdrawal](int(r.beaconCfg.MaxWithdrawalsPerPayload), 44)
+	for _, w := range body.Withdrawals {
+		if w.Index == 0 {
+			return nil, fmt.Errorf("withdrawal index is zero")
+		}
+		ret.Append(&cltypes.Withdrawal{
```

Comment:
is this right? shouldn't you be setting since the entire slice is populated

Reply:
yes it is, it is an empty list, the limit is for hashing